index.js

```diff
@@ -130,91 +143,115 @@ const colorFormatRatio = (color1, color2, convertRatio) => {
   return calculateRatio(RGBColor1, RGBColor2);
 };
 
-const shortToFullHex = (hexColor) => {
-  return [...hexColor].map((x, index) => (index != 0 ? x + x : x)).join("");
+const isValidRGB = (color) => {
+  const rgbRegex = /^rgb\(\s?\d{1,3},\s?\d{1,3},\s?\d{1,3}\)$/i;
+  if (rgbRegex.test(color)) {
+    return rgbInputToRGBNumbers(color).every((v) => 0 <= v && v <= 255);
+  } else return false;
 };
 
-const displayResult = () => {
-  let firstColor = foregroundColor.value;
-  let secondColor = backgroundColor.value;
+const isNotEmpty = (value) => {
+  return value !== null && value !== "";
+};
 
-  const rgbRegex = /^rgb.*/i;
-  const rgbaRegex = /^rgba.*/i;
-  const hexRegex = /^#([A-Fa-f0-9]{6})$/;
-  const hslRegex = /^hsl.*/i;
-  const hslaRegex = /^hsla.*/i;
-  const hexRegex3Digit = /^#[a-fA-F0-9]{3}$/;
-  const isItNamedColor = (color) => namesAndRGBValues.hasOwnProperty(color);
+const showErrorMessage = (div) => {
+  div.classList.remove(`hidden`);
+};
 
+const hideErrorMessage = (div) => {
+  div.classList.add(`hidden`);
+};
+
+const clearErrors = () => {
+  hideErrorMessage(warning);
+  hideErrorMessage(info);
+  ratioResult.innerHTML = "";
+};
+
+const updateSwatchColor = (swatch, color) => {
   if (
-    firstColor.length === 7 ||
-    rgbRegex.test(firstColor) ||
-    rgbaRegex.test(firstColor) ||
-    hslRegex.test(firstColor) ||
-    hslaRegex.test(firstColor) ||
-    hexRegex3Digit.test(firstColor) ||
-    isItNamedColor(firstColor)
-  ) {
-    foregroundSwatch.style.backgroundColor = firstColor;
-  }
-  if (
-    secondColor.length === 7 ||
-    rgbRegex.test(secondColor) ||
-    rgbaRegex.test(secondColor) ||
-    hslRegex.test(secondColor) ||
-    hslaRegex.test(secondColor) ||
-    hexRegex3Digit.test(secondColor) ||
-    isItNamedColor(secondColor)
+    hexRegex.test(color) ||
+    isValidRGB(color) ||
+    rgbaRegex.test(color) ||
+    hslRegex.test(color) ||
+    hslaRegex.test(color) ||
+    hexRegex3Digit.test(color) ||
+    isItNamedColor(color)
   ) {
-    backgroundSwatch.style.backgroundColor = secondColor;
+    swatch.style.backgroundColor = color;
   }
+};
+
+const displayColor = () => {
+  let firstColor = foregroundColor.value;
+  let secondColor = backgroundColor.value;
+  updateSwatchColor(foregroundSwatch, firstColor);
+  updateSwatchColor(backgroundSwatch, secondColor);
+};
+
+const handleChange = () => {
+  clearErrors();
+  displayColor();
+};
+
+const displayResult = () => {
+  let firstColor = foregroundColor.value;
+  let secondColor = backgroundColor.value;
+
   // CASE two Hexes
-  if (firstColor.length === 7 && secondColor.length === 7) {
-    if (hexRegex.test(firstColor) && hexRegex.test(secondColor)) {
+  if (/^#.*/.test(firstColor) && /^#.*/.test(secondColor)) {
```

Comment:
OK. So you nested all hex stuff in the if check, that seemes like good option for now.
Maybe you can create variable for this regexp - but it is totally optional.